cert.py

```diff
@@ -186,4 +199,36 @@ def _get_cert_attribute(oid):
                        critical=False)\
         .sign(ca_key, hashes.SHA256(), default_backend())
 
-    return key, cert
\ No newline at end of file
+    return key, cert
+
+
+# CERTIFICATE SIGNING REQUEST GENERATION
+
+
+def certificate_signing_request(key, country, state, city, org, org_name, common):
+    """
+    Generates a CSR that is based on the passed in key.
+    """
+
+    if not key:
+        # Generate private key here.
+        key, crt = ss_certificate(country, state, city, org, org_name, common)
```

Comment:
It seems like `CertificateSigningRequestBuilder` has no need for the certificate, it generates one itself as part of the CSR. You might modify this function to generate it's own key directly (so that this function is corollary to `ss_certificate()` in that it only requires the attribute values. IOW, don't accept a key, just generate one for the caller, and don't do it via `ss_certificate()` just do it directly.

Reply:
I decided to extract the private key generation to a function of its own. Now `certificate_signing_request` auto-generates a private key instead of allowing one passed in.